api/apps/articles/model/save.coffee

```diff
@@ -68,22 +55,26 @@ removeStopWords = (title) ->
     stoppedTitle = removeStopWords article.thumbnail_title
   slug = _s.slugify article.author?.name + ' ' + stoppedTitle
   return cb null, article if slug is _.last(article.slugs)
+
+  # Append published_at to slug if that slug already exists
   db.articles.count { slugs: slug }, (err, count) ->
     return cb(err) if err
     slug = slug + '-' + moment(article.published_at).format('MM-DD-YY') if count
     article.slugs = (article.slugs or []).concat slug
     cb(null, article)
 
 @generateKeywords = (input, article, cb) ->
+  unless (
+    input.primary_featured_artist_ids or
+    input.featured_artist_ids or
+    input.fair_ids or
+    input.partner_ids or
+    input.contributing_authors or
+    input.tags
+  )
```

Comment:
This is where it's easier to reason that `input` only contains the new fields we want to update.